design.md

```diff
@@ -0,0 +1,84 @@
+# Bazel Module (bzlmod) Support in Renovate
+
+```yaml
+author: Chuck Grindel<chuck.grindel@gmail.com>
+status: To be reviewed
+created: 2023-04-10
+```
+
+## Abstract
+
+[Renovate] is a popular service for updating external dependencies. It [currently supports] updating
+Bazel dependencies described in [WORKSPACE-managed repositories]. With the introduction of [Bazel
+modules], [Renovate] needs to be modified to support the new external dependency mechanism.
+
+## Requirements
+
+- External Dependency Discovery
+  - Support extracting external dependency information from `MODULE.bazel` files.
+    - Read external dependency information from [bazel_dep] declarations.
+    - Support module overrides.
+      - [Single version override]
+      - [Multiple version override]
+      - [Non-registry overrides]
+- Registry Discovery
+  - Support parsing of `.bazelrc` and files imported in the `.bazelrc` looking for [--registry]
+    flags.
+  - If no registry values are detected, default to the [Bazel Central Registry].
+- Bazel Module Release Detection
+  - Detect new module versions when they are introduced to a module registry (e.g., [Bazel Central
+    Registry]).
+    - Triggering updates based upon GitHub releases is problematic as the update to a registry is a
+      separate step that may not complete for minutes to hours (i.e., human approval of a pull
+      request).
+- Versioning
+  - Support [Bazel module version formats] and [compatibility level].
+  - Support [yanked versions]. A version can be yanked by a maintainer when it should be avoided.
+  - Support different version resolution logic for libraries (i.e., is depended upon by other
+    projects) vs executable (i.e., is not depdended upon by other projects) repositories .
+- Bazel Compatibility
+  - Support parsing of `.bazelversion` to detect the Bazel version.
+  - If a Bazel version is present, evaluate the `bazel_compatibility` expressions for the module
+    version to determine if it is an acceptable upgrade candidate.
+
+<!-- Future Sections
+
+## Design
+
+The following sections describe 
+
+### Versioning
+
+#### Parsing and Sorting
+
+#### Resolution: Library vs Executable
```

Comment:
Was a decision made here or is this still up for discussion? IMO, if I'm understanding correctly, this seems like a much bigger project that shouldn't have a bazel specific solution?